### ISSUE TITLE
TypedDict narrowing now too pessamistic?

### ISSUE THREAD

**Issue by anna-intellegens** [microsoft/pyright#10482]
**Describe the bug**
With the new type narrowing changes introduced to fix #10376 it now seems impossible to get the following code to typecheck, and I'm unaware of any other way to narrow unions of `TypedDict`s that aren't tagged that reliably works (which is problematic where we're not in complete control of the schema). The changelog seems to imply that using the proposed `closed` class argument would work, but even adding that failed to let it typecheck.

**Code or Screenshots**
Original, which now fails:
```python
from typing import TypedDict


class TypeA(TypedDict):
    key1: str
    key11: str
    key12: str


class TypeB(TypedDict):
    key2: str
    key21: str


def func(t: TypeA | TypeB) -> str:
    if "key1" in t:
        return t["key11"]
    elif "key2" in t:
        return t["key21"]


if __name__ == "__main__":
    s: str = func({"key1": "hi", "key11": "from", "key12": "me"})

```

But it also fails with the following changes, which was surprising given the changelog message implying it should work:

```python
class TypeA(TypedDict, closed=True): ...
class TypeB(TypedDict, closed=True): ...
```

This also leads to a very confusing error when checking for the key immediately before using it, which is a pattern that works in almost all other situations:

```python
def func2(t: TypeA | TypeB) -> str:
    if "key1" in t and t['key1'] == "ho":
        return "hi"
    else:
        return "no"
```

**VS Code extension or command-line**
Tested with commandline v1.0.401. The code typechecks in v1.0.400

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

It works if you use `closed=True`, but you need to set `enableExperimentalFeatures` to true in your settings to enable the PEP 728 functionality. This PEP is not yet approved, so it could change or be rejected. Until it is approved, pyright's support for it is considered "experimental".
